hip/preconditioner/jacobi_generate_kernel.hip.cpp

```diff
@@ -81,41 +78,11 @@ void generate(syn::value_list<int, max_block_size>,
                   storage_scheme,
               remove_complex<ValueType>* conditioning,
               precision_reduction* block_precisions,
-              const IndexType* block_ptrs, size_type num_blocks)
-{
-    constexpr int subwarp_size = get_larger_power(max_block_size);
-    constexpr int blocks_per_warp = config::warp_size / subwarp_size;
-    const dim3 grid_size(ceildiv(num_blocks, warps_per_block * blocks_per_warp),
-                         1, 1);
-    const dim3 block_size(subwarp_size, blocks_per_warp, warps_per_block);
-
-    if (block_precisions) {
-        hipLaunchKernelGGL(
-            HIP_KERNEL_NAME(
-                kernel::adaptive_generate<max_block_size, subwarp_size,
-                                          warps_per_block>),
-            dim3(grid_size), dim3(block_size), 0, 0, mtx->get_size()[0],
-            mtx->get_const_row_ptrs(), mtx->get_const_col_idxs(),
-            as_hip_type(mtx->get_const_values()), as_hip_type(accuracy),
-            as_hip_type(block_data), storage_scheme, as_hip_type(conditioning),
-            block_precisions, block_ptrs, num_blocks);
-    } else {
-        hipLaunchKernelGGL(
-            HIP_KERNEL_NAME(kernel::generate<max_block_size, subwarp_size,
-                                             warps_per_block>),
-            dim3(grid_size), dim3(block_size), 0, 0, mtx->get_size()[0],
-            mtx->get_const_row_ptrs(), mtx->get_const_col_idxs(),
-            as_hip_type(mtx->get_const_values()), as_hip_type(block_data),
-            storage_scheme, block_ptrs, num_blocks);
-    }
-}
+              const IndexType* block_ptrs, size_type num_blocks);
 
 GKO_ENABLE_IMPLEMENTATION_SELECTION(select_generate, generate);
```

Comment:
It's interesting. The original implementation selection also builds the definition for the practical kernel, but now it gives the usages of the declaration of the kernels.